src/jit/morph.cpp

```diff
@@ -13141,7 +13143,11 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                     // if negative negate (min-int does not need negation)
                     if (mult < 0 && mult != SSIZE_T_MIN)
                     {
-                        tree->gtOp.gtOp1 = op1 = gtNewOperNode(GT_NEG, op1->gtType, op1);
+                        // The type of the new GT_NEG node cannot just be op1->TypeGet().
+                        // Otherwise we may sign-extend incorrectly in cases where the GT_NEG
+                        // node ends up feeding directly a cast, for example in
+                        // GT_CAST<ubyte>(GT_MUL(-1, s_1.ubyte)
```

Comment:
Missing parenthesis at the end of the comment but it's not worth it to rerun ci for this. Let's fix that as part of another PR.

Reply:
Ugh, whoops.  I will make sure to fix that the next time I submit something. Thanks for noticing.